src/main/java/org/generationcp/middleware/dao/WorkbenchUserDAO.java

```diff
@@ -164,26 +168,27 @@ public boolean isUsernameExists(final String username) {
 				for (final WorkbenchUser workbenchUser : workbenchUsers) {
 					final UserDto user = new UserDto();
 					if (workbenchUser.getRoles() != null && !workbenchUser.getRoles().isEmpty()) {
-						// TODO get n roles
-						user.setRole(workbenchUser.getRoles().get(0).getRole());
+						user.setUserRoles(UserRoleMapper.map(workbenchUser.getRoles()));
 					}
 					user.setUserId(workbenchUser.getUserid());
 					if (workbenchUser.getPerson() != null) {
 						user.setEmail(workbenchUser.getPerson().getEmail());
 						user.setFirstName(workbenchUser.getPerson().getFirstName());
 						user.setLastName(workbenchUser.getPerson().getLastName());
+
+						if (workbenchUser.getPerson().getCrops() != null) {
+							final Set<CropDto> crops = new HashSet<>();
+							for (final CropType cropType : workbenchUser.getPerson().getCrops()) {
```

Comment:
You can create `CropMapper `class with this code to follow the pattern of `UserRoleMapper `and `RoleMapper` classes